HumanExposure/factotum
# 2055 Bulk Composition Download fix

Closes #2055 I hope

I was able to trace this problem, in my local environment, to the fact that the celery task itself was not running. The other celery task generated in the same file, `provisional_sid_assignment()`, was running as expected, but the generate_bulk_download_file() was not.

In my troubleshooting, I noticed that when the task is created, the name attribute was set to "generate_bulk_download_csv", rather than "generate_bulk_download_file". I changed this, restarted the server, and the task started running properly.

    sender.add_periodic_task(
        crontab(*settings.GENERATE_BULK_DOWNLOAD_SCHEDULE.split(" ")),
        generate_bulk_download_file.s(),
        name="generate_bulk_download_csv",
    )

I don't know celery well enough to know for sure that this is the source of the problem, but the fact that it's now working for me suggests that it might be. :-)

I'm also not knowledgeable about how celery is deployed on production, so I'm not sure what steps will be necessary to restart the tasks so that it picks up this change.

To Test locally:

Open [project root]/csv_storage, note if there are any files there, and delete them.

Add this cron string to the end of your .env file (you'll want to remove it again when you're done):
GENERATE_BULK_DOWNLOAD_SCHEDULE = * * * * *

From a terminal in the root of the project run:
PS C:\dev\factotum> celery beat -A factotum

Within a minute or so, look in [project root]/csv_storage again and verify that the files have been created.

You should then be able to successfully download the zip file from the Factotum Get Data page.
